### PR TITLE
Collection of smaller changes (Support ESP-IDF 5.3, update docs, update CI pipeline)

### LINKED ISSUE #17: Documentation for ESP32 support
While libcoap-rs now has support for compiling on the ESP32, some documentation that describes the process of setting this up should be added.

Important points:
- necessary sdkconfig.defaults changes (especially for DTLS support)
- version of `esp-idf-sys` in project must match the one specified in libcoap-rs.

FYI:
sdkconfig.defaults:
```
# libcoap
CONFIG_COAP_SERVER_SUPPORT=y
CONFIG_COAP_CLIENT_SUPPORT=y
# enable DTLS in libcoap
CONFIG_COAP_MBEDTLS_PSK=y
CONFIG_COAP_MBEDTLS_PKI=y
CONFIG_MBEDTLS_SSL_PROTO_DTLS=y
CONFIG_MBEDTLS_PSK_MODES=y
CONFIG_MBEDTLS_KEY_EXCHANGE_PSK=y
```

### PR DESCRIPTION
**Pull request by pulsastrix**
This PR:
- Updates the documentation regarding ESP32 support, and building in general
- Adds some additional headers required for ESP-IDF 5.3
- Reworks the CI pipeline to be a bit more informative and generate some badges, which are also shown in the project's `README.md` file.

Closes #17

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
+name: checks
```
Comment (JKRhb) [inline .github/workflows/ci.yml:1]: I think the list of required workflow statuses needs to be adjusted if the name changes here.

How about something like `Build Status` by the way? I think that always looks nice in the GitHub badges.

Comment (pulsastrix) [reply .github/workflows/ci.yml:1]: As we're not only building, but also running the linter and testing, I'd prefer `CI Status` instead of `Build Status`.

Comment (JKRhb) [reply .github/workflows/ci.yml:1]: >  I'd prefer CI Status instead of Build Status

That sounds good :)

**Comment by pulsastrix** [comment]
Frustratingly, some step in transferring the output of the coverage report into the input of the PR comment generation step seems to strip out all newlines.

I'll need to take a deeper look into this tomorrow.

**Review by JKRhb — APPROVED** [review]

**Pull request merged**